### ISSUE TITLE
Allow full configuration of provisioner webhooks in ca.json

### ISSUE THREAD

**Issue by wouttom** [smallstep/certificates#1529]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

We are currently running and configuring step-ca from infra-structure-as code. We want to add a webhook to a provisioner in the configurationfile ca.json.
Currently the entire webhook configuration is read from ca.json EXCEPT the secret that is used to sign the http-requests. The request will get signed by an empty secret if you configure the webhook in ca.json.

Make sure the secret is also read from ca.json. 
To protect it and not store it plaintext/base64 encoded, it might be possible to encrypt the secret with the public key of the intermediate CA. Since the CA has the private key for its normal operations, it can decrypt the secret from the configfile.

## Why is this needed?

To be to manage the configuration of the CA without having to run `step` cli-commands.
This allows for automating the configuration in a predictable way.

**Comment by areed** [comment]
Yes, we wanted to discourage keeping secrets in ca.json. Encrypting with the pub key of the intermediate is an interesting idea, but there may be risks that arise when using that keypair for more than just certificate signing. You can configure the [webhook server to request and verify a client certificate](https://github.com/smallstep/webhooks/blob/main/main.go#L36-L52) if you want to be sure the request is coming from step-ca without using a secret.

**Comment by andsens** [comment]
step-ca is using JWKs to great effect in all other places, why not use one here instead of 3 completely unrelated auth mechanisms? Let users specify a path to a `priv.json` in the `webhooks[]` stanza and apply the same password logic as with the intermediate key.

**Comment by liji-canva** [comment]
btw, the description says `Currently the entire webhook configuration is read from ca.json`, how is this done? I cannot find any document describing configuring webhooks in ca.json. @wouttom

**Comment by wouttom** [comment]
> btw, the description says `Currently the entire webhook configuration is read from ca.json`, how is this done? I cannot find any document describing configuring webhooks in ca.json.

@liji-canva  You can specify the webhook(s) as follows in the provisioner config in ca.json.
```
{
    "type": "SCEP",
    "name": "provisioner",
    "options": {
      "webhooks": [
        {
          "id": "c30dc8c6-7e0d-4fcd-bcf9-b66efbd298c2",
          "name": "webhook",
          "kind": "AUTHORIZING",
          "url": "https://webhook.url/path",
          "certType": "ALL",
          "secret": ""
        }
      ]
    }
  }
```

**Comment by hslatman** [comment]
@liji-canva @wouttom for the record: our current advice w.r.t. configuring webhooks remains, and is to use Remote Management: https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management. 

I understand the use case, and I think it indeed can be useful, but we first need a good solution for managing the secret that works with `ca.json`. And to @andsens point, it would be nice if we didn't have to introduce an entirely new method of working with secrets. 

I'll put this on the board for our next open source triage.

**Comment by ben-hoffman-tanium** [comment]
Just wanted to chime in here and express how much our team would get value from this feature, if there's any updates on it. We are running the helm chart for this project and this is the only thing we can't configure by creating a secret using external-secrets.io

**Comment by spinda** [comment]
I don't think configuring webhooks via `ca.json` even works right now. The `secret` field always ends up as `nil` no matter what you put in the JSON, and the webhook client silently skips the webhook if attempting to deserialize the `secret` value as base64 fails:

https://github.com/smallstep/certificates/blob/3f76d431b63ba8c8238bdaf087026335d0fd3c4d/authority/provisioner/webhook.go#L193-L196

What a waste of debugging time. Why on Earth is that a silent failure?
